File: OPENAPI-REST-API/swagger-client/dart/lib/model/website_table.dart
part of swagger.api;

class WebsiteTable {
  /* Title of the table */
  String title = null;

  List<WebsiteTableRow> rows = [];

  WebsiteTable();

  @override
  String toString() {
    return 'WebsiteTable[title=$title, rows=$rows, ]';
  }

  WebsiteTable.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    title = json['title'];
    rows = WebsiteTableRow.listFromJson(json['rows']);
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'rows': rows
     };
  }

  static List<WebsiteTable> listFromJson(List<dynamic> json) {
    return json == null ? new List<WebsiteTable>() : json.map((value) => new WebsiteTable.fromJson(value)).toList();
  }

  static Map<String, WebsiteTable> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, WebsiteTable>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new WebsiteTable.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/WebsiteTable.cpp


#include "WebsiteTable.h"

using namespace Tiny;

WebsiteTable::WebsiteTable()
{
	title = std::string();
	rows = std::list<WebsiteTableRow>();
}

WebsiteTable::WebsiteTable(std::string jsonString)
{
	this->fromJson(jsonString);
}

WebsiteTable::~WebsiteTable()
{

}

void
WebsiteTable::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *titleKey = "title";

    if(object.has_key(titleKey))
    {
        bourne::json value = object[titleKey];



        jsonToValue(&title, value, "std::string");


    }

    const char *rowsKey = "rows";

    if(object.has_key(rowsKey))
    {
        bourne::json value = object[rowsKey];


        std::list<WebsiteTableRow> rows_list;
        WebsiteTableRow element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            rows_list.push_back(element);
        }
        rows = rows_list;


    }


}

bourne::json
WebsiteTable::toJson()
{
    bourne::json object = bourne::json::object();





    object["title"] = getTitle();





    std::list<WebsiteTableRow> rows_list = getRows();
    bourne::json rows_arr = bourne::json::array();

    for(auto& var : rows_list)
    {
        WebsiteTableRow obj = var;
        rows_arr.append(obj.toJson());
    }
    object["rows"] = rows_arr;




    return object;

}

std::string
WebsiteTable::getTitle()
{
	return title;
}

void
WebsiteTable::setTitle(std::string title)
{
	this->title = title;
}

std::list<WebsiteTableRow>
WebsiteTable::getRows()
{
	return rows;
}

void
WebsiteTable::setRows(std::list<WebsiteTableRow> rows)
{
	this->rows = rows;
}



